src/types.rs

```diff
@@ -577,38 +617,46 @@ impl Term {
             }
         }
     }
-    /// Every [`Atom`] used in the term.
+    /// Returns every [`Atom`] used in the term.
     ///
     /// [`Atom`]: enum.Atom.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Atom};
+    /// let mut sig = Signature::default();
+    /// let var = sig.new_var(Some("Z".to_string()));
+    /// let op = sig.new_op(0, Some("A".to_string()));
+    /// let args = vec![];
+    ///
+    /// let var_term = Term::Variable(var);
+    /// let op_term = Term::Application{op, args};
+    ///
+    /// assert_eq!(var_term.atoms(), vec![Atom::Variable(var)]);
+    /// assert_eq!(op_term.atoms(), vec![Atom::Operator(op)]);
+    /// ```
     pub fn atoms(&self) -> Vec<Atom> {
         let vars = self.variables().into_iter().map(Atom::Variable);
         let ops = self.operators().into_iter().map(Atom::Operator);
         vars.chain(ops).collect()
     }
-    /// Every [`Variable`] used in the term.
+    /// Returns every [`Variable`] used in the `Term`.
     ///
     /// [`Variable`]: struct.Variable.html
     ///
     /// # Examples
     ///
     /// ```
-    /// # use term_rewriting::{Signature, parse_term,Term};
+    /// # use term_rewriting::{Signature, parse_term, Term};
     /// let mut sig = Signature::default();
-    /// let a = sig.new_op(2, Some(".".to_string()));
-    /// let s = sig.new_op(0, Some("S".to_string()));
-    /// let k = sig.new_op(0, Some("K".to_string()));
     ///
     /// let t = parse_term(&mut sig, "S K y_ z_").expect("parse of S K y_ z_");
-    /// let t2 = parse_term(&mut sig, "S K y_ z_").expect("parse of S K y_ z_");
-    /// let t3 = parse_term(&mut sig, "S K y_").expect("parse of S K y_");
     ///
     /// let vars = t.variables();
-    /// let vars2 = t2.variables();
-    /// let vars3 = t3.variables();
+    /// let var_names: Vec<String> = vars.iter().map(|v| v.display(&sig)).collect();
     ///
-    /// assert_ne!(vars,vars2);
-    /// assert_ne!(vars2,vars3);
-    /// assert_ne!(vars,vars3);
+    /// assert_eq!(var_names, vec!["y".to_string(), "z".to_string()]);
     /// ```
```

Comment:
Double-check this test. I think it fails against the most recent version of `display`.